tests/test_interpreter.py

```diff
@@ -228,6 +226,89 @@ def test_read_line(self):
         result = self.run_code_with_input(code, "Alice\n")
         self.assertEqual(result, "Alice")
 
+    def test_for_loop(self):
+        result = self.run_code("""
+            (define sum 0)
+            (for i 0 4
+                (set sum (+ sum 1))
+            )
+            sum
+        """)
+        self.assertEqual(result, 4)
+
+    def test_get_seq_with_list(self):
+        result = self.run_code("""
+            (define xs (list 10 20 30))
+            (get-seq xs 1)
+        """)
+        self.assertEqual(result, 20)
+
+    def test_get_seq_with_string(self):
+        result = self.run_code("""
+            (define s "hello")
+            (get-seq s 1)
+        """)
+        self.assertEqual(result, "e")
+
+    def test_get_seq_out_of_bounds(self):
+        with self.assertRaises(RuntimeError):
+            self.run_code("""
+                (define xs (list 1 2))
+                (get-seq xs 5)
+            """)
+
+    def test_get_seq_wrong_type(self):
+        with self.assertRaises(RuntimeError):
+            self.run_code("""
+                (define x 123)
+                (get-seq x 0)
+            """)
+
+    def test_set_seq_variable(self):
+        result = self.run_code("""
+            (define data (list 1 2 3))
+            (set-seq data (list 7 8 9))
```

Comment:
I don't really get `set-seq`.  `get-seq` gets a value at an index.  What's the difference between `set-seq` and `set`? Why have both?